autogpt_platform/frontend/src/lib/utils.ts

```diff
@@ -312,3 +312,48 @@ export function findNewlyAddedBlockCoordinates(
     y: 0,
   };
 }
+
+type ParsedKey = { key: string; index?: number };
+
+export function parseKeys(key: string): ParsedKey[] {
+  const splits = key.split(/_@_|_#_|_\$_|\./);
+  const keys: ParsedKey[] = [];
+  let currentKey: string | null = null;
+
+  splits.forEach((split) => {
+    const isInteger = /^\d+$/.test(split);
+    if (!isInteger) {
+      if (currentKey !== null) {
+        keys.push({ key: currentKey });
+      }
+      currentKey = split;
+    } else {
+      if (currentKey !== null) {
+        keys.push({ key: currentKey, index: parseInt(split, 10) });
+        currentKey = null;
+      } else {
+        throw new Error("Invalid key format: array index without a key");
+      }
+    }
+  });
+
+  if (currentKey !== null) {
+    keys.push({ key: currentKey });
+  }
+
+  return keys;
+}
```

Comment:
we should probably add security test for this lol

Reply:
I agree but this code is already in the codebase I just moved it!